test/end2end_test.go

```diff
@@ -7032,6 +7032,84 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 	}()
 }
 
+// TestClientTransportRestartsAfterStreamIdExhausted tests that the client transport
+// drains and restarts when next stream ID exceeds MaxStreamID. This test also
+// verifies that subsequent RPCs use a new client transport and the old transport
+// is closed.
+func (s) TestClientTransportRestartsAfterStreamIdExhausted(t *testing.T) {
+	// overriding MaxStreamIDForTesting.
```

Comment:
There's a lot going on here:

- This is not a complete sentence, so it shouldn't really end with a "."
- Generally we'd use an imperative (command) verb like `// Override MaxStreamIDForTesting.`  (Like this it actually would be a complete sentence with the implied "You" subject.)
- This comment isn't that useful since the code very clearly overrides MaxStreamIDForTesting.

Comments should generally do more than say _what_ the code is doing; they should say _why_.  (If a block of code is really complicated, it can be useful to say what it is doing.)

For something that's obvious, not having any comments is better than having comments that point out the obvious.

If you wanted a comment here, I'd write it as: `// Set the transport's MaxStreamID to 5 to cause connections to drain after 2 RPCs.` But I'll leave it up to you to decide whether to comment at all.